core/src/ethereum/EthereumLikeExtendedPublicKey.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ *
+ * EthereumLikeExtendedPublicKey
+ * ledger-core
+ *
+ * Created by El Khalil Bellakrid on 08/07/2018.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include "EthereumLikeExtendedPublicKey.h"
+#include "EthereumLikeAddress.h"
+
+#include <math/Base58.hpp>
+
+#include <bytes/BytesReader.h>
+#include <bytes/BytesWriter.h>
+
+#include <utils/Exception.hpp>
+
+#include <crypto/Keccak.h>
+#include <crypto/SECP256k1Point.hpp>
+#include <crypto/RIPEMD160.hpp>
+#include <crypto/SHA256.hpp>
+#include <crypto/HashAlgorithm.h>
+
+namespace ledger {
+    namespace core {
+
+        EthereumLikeExtendedPublicKey::EthereumLikeExtendedPublicKey(const api::Currency& params,
+                                                                     const DeterministicPublicKey& key,
+                                                                     const DerivationPath& path):
+            _currency(params), _key(key), _path(path)
+        {
+
+        }
+
+        static inline DeterministicPublicKey _derive(int index, const std::vector<uint32_t>& childNums, const DeterministicPublicKey& key) {
```

Comment:
This is due to the fact that we changed this model (derivation or not) several times because Vault and Live team were not agreeing on same one, but indeed we can remove it for the moment ...